src/client/datascience/jupyter/kernels/kernelCommandListener.ts

```diff
@@ -142,12 +151,47 @@ export class KernelCommandListener implements IDataScienceCommandListener {
                 );
                 if (response === dontAskAgain) {
                     await this.disableAskForRestart(document.uri);
-                    void kernel.restart();
+                    void this.wrapKernelMethod('restart', document.uri, kernel, () => kernel.restart());
                 } else if (response === yes) {
-                    void kernel.restart();
+                    void this.wrapKernelMethod('restart', document.uri, kernel, () => kernel.restart());
                 }
             } else {
-                void kernel.restart();
+                void this.wrapKernelMethod('restart', document.uri, kernel, () => kernel.restart());
+            }
+        }
+    }
+
+    private async wrapKernelMethod(
+        context: 'interrupt' | 'restart',
+        resource: Resource,
+        kernel: IKernel,
+        method: () => Promise<void>
+    ) {
+        // Get currently executing cell
+        const currentCell = kernel.pendingCells[0];
+        try {
+            await method();
+        } catch (ex) {
+            // Translate the result
+            const result = await this.errorHandler.handleKernelError(
+                ex,
+                context,
+                kernel.kernelConnectionMetadata,
+                resource
+            );
+
+            switch (result.kind) {
+                case 'Canceled':
+                    ex = new CancellationError(
+                        DataScience.canceledKernelHeader().format(
+                            kernel.kernelConnectionMetadata.interpreter?.displayName || ''
+                        )
+                    );
+                    break;
+            }
+            if (currentCell) {
+                const cellExecution = CellExecutionCreator.getOrCreate(currentCell, kernel.controller);
+                displayErrorsInCell(currentCell, cellExecution, ex).ignoreErrors();
```

Comment:
Looking at the code in `displayErrorsInCell` we could end up dumping the error object as is, as we're expecting a string as the 3rd argument.

Reply:
I believe you're mistaken. ex.toString does not include the stack. Just the message.

Reply:
Yes, you are right, thanks

Reply:
Sorry